app/pluginmanager.js

```diff
@@ -1105,67 +1105,70 @@ PluginManager.prototype.unzipPackage = function () {
 
 PluginManager.prototype.checkPluginDependencies = function (folder) {
   var self = this;
-  var pendingResult = libQ.resolve();
-  
-  var result = { success: 'failed', message: 'Plugin failed the dependency check', folder: folder};
+  var pendingResult = libQ.resolve({ success: 'success', message: '', folder: folder});
   
   self.logger.info('Check plugin dependencies');
 
   var package_json = self.getPackageJson(folder);
   
-  var nodeCheck = false;
-  var volumioCheck = false;
-  var message = '';
   if(package_json.engines) {
     
     pendingResult = pendingResult
-    .then(() => {
+    .then((result) => {
         if(package_json.engines.node) {
-            nodeCheck = true;
+            result.nodeCheck = true;
             var nodeVersion = semver.coerce(process.versions.node);
-            if(!semver.satisfies(nodeVersion, package_json.engines.node)) {
-                message += 'Node version ' + nodeVersion + ' not usable with plugin. ';
+            if(nodeVersion === null) {
+                result.success = 'warn';
+                result.message += 'Current Node version cannot be detected. ';
+            } else if(!semver.satisfies(nodeVersion, package_json.engines.node)) {
+                result.success = 'failed';
+                result.message += 'Node version ' + nodeVersion + ' not usable with plugin. ';
             }
         }
-    })
-    .then(() => {    
+        
         if(package_json.engines.volumio) {
-            volumioCheck = true;
+            result.volumioCheck = true;
             return self.coreCommand.getSystemVersion()
                 .then(e => {
-                    var volumioVersion = semver.coerce(e.systemversion);
-                    if(!semver.satisfies(volumioVersion, package_json.engines.volumio)) {
-                        message += 'Volumio version ' + volumioVersion + ' not usable with plugin. ';
+                    var volumioVersion = semver.coerce(e.systemversion, { loose: true });
+                    if(nodeVersion === null) {
```

Comment:
Should be `volumioVersion === null` no?

Reply:
Good spot - I was attempting to work quickly and clearly missed a copy/paste. I've pushed another commit which should fix this, although the `{loose: true}` should make the null a very rare occurrence.